Automattic/wp-calypso
# Store Services: Removing redundant commas when state is missing from an address

#### Changes proposed in this Pull Request

* `<AddressSummary />` is now smarter about displaying addresses and does not add redundant commas if there is no state (the other fields are mandatory).
* Introduction of the `isEqualCaseInsensitive` function, which replaces `isEqual` in the `submitAddressForNormalization` action in order to allow for the silent replacement of addresses when the only thing that changes is capitalization.
* The removal of the `success` flag/parameter of `handleNormalizeResponse` in order to be able to automatically collapse the address step. This change is already implemented in https://github.com/Automattic/wp-calypso/pull/28533 (merged), but https://github.com/Automattic/wp-calypso/pull/28593 (base branch of this one) still does not contain it.

[![](https://cdn-std.dprcdn.net/files/acc_693496/VIIQry)](https://cld.wthms.co/VIIQry)

#### Testing instructions

1. Start creating a new shipping label
2. Enter a UK address without entering a state
3. See the suggestion that includes the state. There should be no comma before the postal code neither in the original address nor the suggested one.

Fixes https://github.com/Automattic/woocommerce-services/issues/1491
